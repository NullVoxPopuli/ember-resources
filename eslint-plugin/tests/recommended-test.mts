import { it, describe, expect } from 'vitest';

import { rules } from '../src/index.js';

describe('recommended rules', () => {
  it('has the right list', () => {
    const errors: string[] = [];

    Object.keys(rules).forEach((name) => {
      if (rules[name]?.meta?.docs?.recommended) {
        errors.push(name);
      }
    });

    expect(errors).toMatchSnapshot();
  });
});
