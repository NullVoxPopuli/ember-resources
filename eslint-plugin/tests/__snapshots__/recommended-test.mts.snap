// Vitest Snapshot v1

exports[`recommended rules > has the right list 1`] = `
[
  "no-this-property-assignment-in-resource-thunk",
  "no-this-property-assignment-in-tracked-function",
  "no-this-property-assignment-in-function-resource",
]
`;
